centreon/src/Core/AgentConfiguration/Infrastructure/Repository/DbReadAgentConfigurationRepository.php

```diff
@@ -0,0 +1,214 @@
+    {
+        $sql = <<<'SQL'
+            SELECT
+                rel.`poller_id` as id,
+                ng.`name`
+            FROM `:db`.`ac_poller_relation` rel
+            JOIN `:db`.`agent_configuration` ac
+                ON rel.ac_id = ac.id
+            JOIN `:db`.`nagios_server` ng
+                ON rel.poller_id = ng.id
+            WHERE ac.`type` = :type
+            SQL;
+
+        // Prepare SQL + bind values
+        $statement = $this->db->prepare($this->translateDbName($sql));
+        $statement->bindValue(':type', $type->value, \PDO::PARAM_STR);
+        $statement->setFetchMode(\PDO::FETCH_ASSOC);
+        $statement->execute();
+
+        // Retrieve data
+        $pollers = [];
+        foreach ($statement as $result) {
+            /** @var array{id:int,name:string} $result */
+            $pollers[] = new Poller($result['id'], $result['name']);
+        }
+
+        return $pollers;
+    }
+
+    /**
+     * Find all the pollers associated with an AC ID.
+     *
+     * @param int $agentConfigurationId
+     *
+     * @throws \Throwable
+     *
+     * @return Poller[]
+     */
+    public function findPollersByAcId(int $agentConfigurationId): array
+    {
+        $statement = $this->db->prepare($this->translateDbName(
+            <<<'SQL'
+                SELECT
+                    rel.`poller_id` as id,
+                    ng.`name`
+                FROM `:db`.`ac_poller_relation` rel
+                JOIN `:db`.`nagios_server` ng
+                    ON rel.poller_id = ng.id
+                WHERE rel.`ac_id` = :id
+                SQL
+        ));
+        $statement->bindValue(':id', $agentConfigurationId, \PDO::PARAM_INT);
+        $statement->setFetchMode(\PDO::FETCH_ASSOC);
+        $statement->execute();
+
+        // Retrieve data
+        $pollers = [];
+        foreach ($statement as $result) {
+            /** @var array{id:int,name:string} $result */
+            $pollers[] = new Poller($result['id'], $result['name']);
+        }
+
+        return $pollers;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findPollersWithBrokerModuleDirective(string $module): array
+    {
+        $statement = $this->db->prepare($this->translateDbName(
+            <<<'SQL'
+                SELECT
+                    `cfg_nagios_id`
+                FROM `:db`.`cfg_nagios_broker_module`
+                WHERE `broker_module` = :module
+                SQL
+        ));
+        $statement->bindValue(':module', $module, \PDO::PARAM_STR);
+        $statement->execute();
+
+        return $statement->fetchAll(\PDO::FETCH_COLUMN);
+    }
+
+    /**
+     * @param _AgentConfiguration $row
+     *
+     * @return AgentConfiguration
+     */
+    private function createFromArray(array $row): AgentConfiguration
+    {
+        /** @var array<string,mixed> $configuration */
+        $configuration = json_decode(json: $row['configuration'], associative: true, flags: JSON_OBJECT_AS_ARRAY);
```

Comment:
you should isset the key "configuration", and beware json_decode may return null or you can use the flag JSON_THROW_ON_ERROR to throw an exception

Reply:
phpstan doesn't like the isset, but I added the flag

Reply:
it's ok !